build/teamcity/internal/cockroach/build/ci/open-new-bazel-builder-image-pr.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+set -xeuo pipefail
+
+# This script expects the following env vars to be set:
+#   BRANCH [branch name to be created in fork]
+#   VERSION [new version to echo into build/.bazelbuilderversion]
+#   GH_TOKEN [github token]
+
+dir="$(dirname $(dirname $(dirname $(dirname $(dirname $(dirname "${0}"))))))"
+source "$dir/teamcity-common-support.sh"
+
+git_ssh() {
+  # $@ passes all arguments to this function to the command
+  GIT_SSH_COMMAND="ssh -i /tmp/.cockroach-teamcity-key" git "$@"
+}
+
+# Configure git.
+git config --global user.email "teamcity@cockroachlabs.com"
+git config --global user.name "cockroach-teamcity"
+configure_git_ssh_key
+WORKDIR="$(mktemp -d ./workdir.XXXXXX)"
+mv .cockroach-teamcity-key $WORKDIR/.cockroach-teamcity-key
+
+git_ssh clone "ssh://git@github.com/cockroachdb/cockroach.git" "$WORKDIR/cockroach" && cd "$WORKDIR/cockroach"
```

Comment:
I wonder if you even need to clone the repo. 🤔 TeamCity already has it. Can't you just reuse it? Or you need to have the checkout on a different commit?

Reply:
It's a little risky. When I was testing from my PR branch and want to go to master I was getting vendor diffs, etc.. so decided to increase the time by a minute by cloning and work from a clean workspace instead.